csi/src/mount.rs

```diff
@@ -111,58 +111,97 @@ pub fn mount_opts_compare(m1: &[String], m2: &[String], ro: bool) -> bool {
 }
 
 // Return supported filesystems and their default mount options.
-pub fn probe_filesystems() -> Result<Vec<Fs>, String> {
+pub fn probe_filesystems() -> Vec<Fs> {
     let mut filesystems = Vec::new();
     // the first filesystem is the default one
     let supported_fs = ["xfs", "ext4"];
+
+    for fsname in supported_fs.iter() {
+        match probe_filesystem(fsname) {
+            Ok(opts) => filesystems.push(Fs {
+                name: (*fsname).to_string(),
+                defaults: opts,
+            }),
+            Err(err) => {
+                warn!("Filesystem {} will not be available: {}", fsname, err)
+            }
+        }
+    }
+
+    filesystems
+}
+
+fn probe_filesystem(fsname: &str) -> Result<Vec<String>, String> {
+    let mi = MountInfo {
+        source: format!("/tmp/fs-{}.img", fsname),
+        dest: format!("/tmp/fs_default_{}", fsname),
+        opts: Vec::new(),
+    };
+    prepare_mount(&mi)?;
+
+    match probe_defaults(fsname, &mi) {
+        Ok(opts) => {
+            cleanup_mount(&mi).unwrap_or_else(|e| {
+                warn!("{}", e);
+            });
+            Ok(opts)
+        }
+        Err(err) => Err(format!(
+            "Filesystem {} will not be available: {}",
```

Comment:
it seems that the same message will be printed twice (in probe_filesystems you have the same). I would say that `let opts = probe_defaults(fsname, &mi)?;` would work just fine.

Reply:
Oh, yes - didn't notice that...  fixing.